Tools/WinMLRunner/src/EventTraceHelper.cpp

```diff
@@ -0,0 +1,567 @@
+        dataDescriptor.PropertyName = (ULONGLONG)((PBYTE)(pInfo) + pInfo->EventPropertyInfoArray[j].NameOffset);
+        dataDescriptor.ArrayIndex = ULONG_MAX;
+        status = TdhGetPropertySize(pEvent, 0, NULL, 1, &dataDescriptor, &propertySize);
+        status = TdhGetProperty(pEvent, 0, NULL, 1, &dataDescriptor, propertySize, (PBYTE)&Length);
+        *PropertyLength = (USHORT)Length;
+    }
+    else
+    {
+        if (pInfo->EventPropertyInfoArray[i].length > 0)
+        {
+            *PropertyLength = pInfo->EventPropertyInfoArray[i].length;
+        }
+        else
+        {
+            // If the property is a binary blob and is defined in a MOF class, the extension
+            // qualifier is used to determine the size of the blob. However, if the extension
+            // is IPAddrV6, you must set the PropertyLength variable yourself because the
+            // EVENT_PROPERTY_INFO.length field will be zero.
+
+            if (TDH_INTYPE_BINARY == pInfo->EventPropertyInfoArray[i].nonStructType.InType &&
+                TDH_OUTTYPE_IPV6 == pInfo->EventPropertyInfoArray[i].nonStructType.OutType)
+            {
+                *PropertyLength = (USHORT)sizeof(IN6_ADDR);
+            }
+            else if (TDH_INTYPE_UNICODESTRING == pInfo->EventPropertyInfoArray[i].nonStructType.InType ||
+                     TDH_INTYPE_ANSISTRING == pInfo->EventPropertyInfoArray[i].nonStructType.InType ||
+                     (pInfo->EventPropertyInfoArray[i].Flags & PropertyStruct) == PropertyStruct)
+            {
+                *PropertyLength = pInfo->EventPropertyInfoArray[i].length;
+            }
+            else
+            {
+                printf("Unexpected length of 0 for intype %d and outtype %d\n",
+                        pInfo->EventPropertyInfoArray[i].nonStructType.InType,
+                        pInfo->EventPropertyInfoArray[i].nonStructType.OutType);
+
+                status = ERROR_EVT_INVALID_EVENT_DATA;
+            }
+        }
+    }
+
+    return status;
+}
+
+static DWORD GetEventInformation(PEVENT_RECORD pEvent, PTRACE_EVENT_INFO& pInfo)
+{
+    DWORD status = ERROR_SUCCESS;
+    DWORD bufferSize = 0;
+
+    // Retrieve the required buffer size for the event metadata.
+
+    status = TdhGetEventInformation(pEvent, 0, NULL, pInfo, &bufferSize);
+
+    if (ERROR_INSUFFICIENT_BUFFER == status)
+    {
+        pInfo = (TRACE_EVENT_INFO*)malloc(bufferSize);
+        if (pInfo == NULL)
+        {
+            printf("Failed to allocate memory for event info (size=%lu).\n", bufferSize);
+            status = ERROR_OUTOFMEMORY;
+            return status;
+        }
+
+        // Retrieve the event metadata.
+
+        status = TdhGetEventInformation(pEvent, 0, NULL, pInfo, &bufferSize);
+    }
+
+    if (ERROR_SUCCESS != status)
+    {
+        printf("TdhGetEventInformation failed with 0x%x.\n", status);
+    }
+
+    return status;
+}
+
+static std::wstring& TrimOperatorName(std::wstring& s)

```

Comment:
>TrimOperatorName [](http://example.com/codeflow?start=21&length=16)

A little odd that this helper is destructive

Reply:
By destructive you mean it modifies the variable? this is a helper method that modifies an intermediate variable, should be fine?

Reply:
He just means that it was odd to return a reference to the modified string rather than a new instance. Typically a function either returns a new string or modifies it in place but returns nothing. e.g. either a or b

a. `void TrimOperatorName(std:;string& s)`
b. `std::string TrimOperatorName(std:;string_view s)`

Reply:
hmmmm, modified the method to return a new instance instead of reference